src/cable/simulation/cable_simul_addr.cpp

```diff
@@ -1,35 +1,24 @@
 #include "cable/simulation/cable_simul_addr.hpp"
 
 
-c_cable_simul_addr::c_cable_simul_addr(int uuid) : c_cable_base_addr(uuid) { }
+c_cable_simul_addr::c_cable_simul_addr(int uuid) : c_cable_base_addr(t_cable_kind::kind_simul),
+m_addr(uuid)
+{ }
 
-void c_cable_simul_addr::print(ostream & ostr) const {
-	auto uuid = boost::any_cast<int>(get_addrdata());
-	ostr << "[transp-simul-" << uuid << "]" ;
-}
-
-std::string c_cable_simul_addr::cable_type_name() const {
-	return "simulation"s;
+unique_ptr<c_cable_base_addr> c_cable_simul_addr::clone() const { ///< polymorphic clone
+	return make_unique< c_cable_simul_addr >( m_addr );
```

Comment:
@robertoleksy yeap m_addr is just an object (not a pointer/smartpointer), here of type int, so we just use it by value to set the value of newly allocated object - allocated by make_unique